engine/storage/volume/src/main/java/org/apache/cloudstack/storage/datastore/provider/DefaultHostListener.java

```diff
@@ -55,8 +57,12 @@
 import javax.inject.Inject;
 import java.util.List;
 
-public class DefaultHostListener implements HypervisorHostListener {
+public class DefaultHostListener implements HypervisorHostListener, Configurable {
     private static final Logger s_logger = Logger.getLogger(DefaultHostListener.class);
+    ConfigKey<Integer> ModifyStoragePoolCommandWait = new ConfigKey<Integer>("Advanced", Integer.class,
+            "modify.storage.pool.command.wait", "60",
+            "Time in seconds to wait for ModifyStoragePoolCommand command to return", true);
```

Comment:
CS have lot of such cmds, maybe not good idea to add a config at cmd level. (later, might end up one config for each cmd). generic configs, eg. storage.command.wait / host.command.wait, based on the category with appropriate descriptions might be good.

Reply:
I agree with you. But generic config might not work. e.g. `storage.command.wait` as 60 seconds will work for `ModifyStoragePoolCommand` but not might be appropriate for other storage related commands. It needs to be specific to a `Command`.

IMO, it would be better to have something which allows the user to set configuration if required like `command.<command name>.wait` and wait is set only if the value is set. If it's not set, then some default value is used.

Reply:
What about
agent.startup.command.timeout
agent.startup.script.timeout 
?